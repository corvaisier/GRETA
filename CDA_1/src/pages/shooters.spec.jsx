import React from 'react'
import { render, fireEvent, screen, act } from '@testing-library/react'

import Shooter from './shooters'

describe('<Shooters />', () => {
    it('Should display error message for negative number', async () => {
        render(<Shooter />)
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "-3" }} );
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        })
        screen.getByText("Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99")
        expect(document.body).toMatchSnapshot();
    })
    it('Should display an error message for numbers greater than 99', async () => {
        render(<Shooter />)
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "101" }} )
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        })
        screen.getByText("Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99")
        expect(document.body).toMatchSnapshot();
    })
    it('Should display an error message for anything else than numbers', async () => {
        render(<Shooter />)
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "AHA!" }} )
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        })
        screen.getByText("Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99")
        expect(document.body).toMatchSnapshot();
    })
    it('Should display the song of zero shooters if value equal 0', async () => {
        render(<Shooter />)
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "0" }} )
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        })
        screen.getByText("Plus de shooters sans alcool sur le mur, plus de shooters sans alcool. Vas au supermarché pour en acheter, 99 shooters sans alcool sur le mur.")
        expect(document.body).toMatchSnapshot();
    })
    it('Should display the song of zero shooters if value equal 1', async () => {
        render(<Shooter />)
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "1" }} )
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        })
        screen.getByText("1 shooter sans alcool sur le mur, 1 shooter sans alcool. Bois en un et au suivant ! plus de shooters sans alcool sur le mur.")
        expect(document.body).toMatchSnapshot();
    })
    it('Should display the song of zero shooters if value equal 2', async () => {
        render(<Shooter />)
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "2" }} )
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        })
        screen.getByText("2 shooters sans alcool sur le mur, 2 shooters sans alcool. Bois en un et au suivant! 1 shooter sans alcool sur le mur.")
        expect(document.body).toMatchSnapshot();
    })
})