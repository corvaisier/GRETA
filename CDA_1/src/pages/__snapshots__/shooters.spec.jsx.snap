// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shooters /> Should display an error message for anything else than numbers 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin-top: 30px;"
    >
      <form
        class="form-group"
      >
        <label>
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="AHA!"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div>
          Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99
        </div>
      </form>
    </div>
  </div>
</body>
`;

exports[`<Shooters /> Should display an error message for numbers greater than 99 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin-top: 30px;"
    >
      <form
        class="form-group"
      >
        <label>
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="101"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div>
          Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99
        </div>
      </form>
    </div>
  </div>
</body>
`;

exports[`<Shooters /> Should display error message for negative number 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin-top: 30px;"
    >
      <form
        class="form-group"
      >
        <label>
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="-3"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div>
          Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99
        </div>
      </form>
    </div>
  </div>
</body>
`;

exports[`<Shooters /> Should display the song of zero shooters if value equal 0 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin-top: 30px;"
    >
      <form
        class="form-group"
      >
        <label>
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="0"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div />
      </form>
      <p>
        Plus de shooters sans alcool sur le mur, plus de shooters sans alcool. Vas au supermarché pour en acheter, 99 shooters sans alcool sur le mur.
      </p>
    </div>
  </div>
</body>
`;

exports[`<Shooters /> Should display the song of zero shooters if value equal 1 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin-top: 30px;"
    >
      <form
        class="form-group"
      >
        <label>
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="1"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div />
      </form>
      <p>
        1 shooter sans alcool sur le mur, 1 shooter sans alcool. Bois en un et au suivant ! plus de shooters sans alcool sur le mur.
      </p>
    </div>
  </div>
</body>
`;

exports[`<Shooters /> Should display the song of zero shooters if value equal 2 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin-top: 30px;"
    >
      <form
        class="form-group"
      >
        <label>
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="2"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div />
      </form>
      <p>
        2 shooters sans alcool sur le mur, 2 shooters sans alcool. Bois en un et au suivant! 1 shooter sans alcool sur le mur.
      </p>
    </div>
  </div>
</body>
`;
